src/app/settings/categories/regressiontypes/regressiontypes.component.ts

```diff
@@ -99,13 +98,19 @@ export class RegressionTypesComponent implements OnInit, OnDestroy {
         }
         this._settingsservice
             .getEntities(this.configSettings.nssBaseURL + this.configSettings.regTypeURL+"?regions="+ this.selectedRegionID +"&statisticgroups="+ this.selectedStatisticID)
-            .subscribe(regs => {
-                this.regressionTypes = regs;
+            .subscribe(res => {
+                this.nssRegressionTypes = res;
+            });
+        this._settingsservice
+            .getEntities(this.configSettings.gageStatsBaseURL + this.configSettings.regTypeURL+"?regions="+ this.selectedRegionID +"&statisticgroups="+ this.selectedStatisticID)
+            .subscribe(res => {
+                this.gsRegressionTypes = res;
+                this.combineRegressionTypes();
```

Comment:
So, my only thought here is that this might cause issues if the gagestats service calls come back before the nss calls. In that case, we might not be combining things correctly. What we should do here is have a check for whether the other response has come back and if it has then call `combineRegressionTypes`. We'll want to do that when we call the nss and gagestats regression types, so, in the nss call, if the gagestats reg types have come back, then call combine.

Hope that makes sense. We'll want to replicate this check (a) when the filters trigger new calls to the services and (b) in the other settings pages that combine nss and gagestats calls as well.